SYS-1827: Upgrade Django, Python, and PostgreSQL

ztucker4 opened the pull request:
Implements [SYS-1827](https://uclalibrary.atlassian.net/browse/SYS-1827)

Upgrades Python to 3.13, Django to 5.2.1, and PostgreSQL to 16. 

**Note:** several packages in `requirements.txt` were updated to support the new versions of Django and Python. `pandas` is among these - 2.2.3, the latest release, is the first to support Python 3.13. Before now, we had pinned `numpy` at an older version (see [comment in current requirements.txt](https://github.com/UCLALibrary/LBS/blob/e4efdad704a9888803dd15f92b2959cb16019a6a/requirements.txt#L7)), which is not compatible with `pandas` 2.2.3. I'm not sure why this was needed at the time. I've run a few test reports and the resulting Excel files look ok to me, but I don't know exactly what to look for. 

In addition to updating the Logout links to send POST requests, some CSS and HTML changes were needed to fix the styling of the navbar in the QDB interface.

### Testing

Remove old volumes, if needed
`docker volume rm lbs_pg_data`
`docker volume rm lbs_pgadmin_data`  

Build and run containers (including separate pgAdmin container)
`docker compose build`
`docker compose up -d`
`docker compose -f docker-compose_PGADMIN.yml up -d`

Run tests - 84, all passing
`docker compose exec django python manage.py test`

Check out the application in the browser: http://localhost:8000/admin/.
Check out pgAdmin in the browser:  http://localhost:5050/ (additional setup instructions in the [README](https://github.com/UCLALibrary/LBS/blob/main/README.md))

[SYS-1827]: https://uclalibrary.atlassian.net/browse/SYS-1827?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

akohler commented:
@ztucker4 The `numpy` requirement was specific to the version of `pandas` we were using at the time, [2.1.1](https://github.com/pandas-dev/pandas/blob/a671b5a8bf5dd13fb19f0e88edc679bc9e15c673/pyproject.toml#L13)
`    "numpy>=1.26.0,<2; python_version>='3.12'",`
Now they just want `numpy` 2.0 or greater:
`"numpy>=2.0.0rc1",`
, so removing our explicit requirement is fine.

akohler reviewed (changes requested):
@ztucker4 One `logout` reference still needs updating: the one on the [QDB Reports](http://localhost:8000/qdb/report/) page is a direct link to http://localhost:8000/qdb/logout/
That appears to come from `qdb/templates/base.html`.

Also, please make the code changes mentioned in the emails with Alli, as part of this PR.  Once they're deployed, I'll update the data itself as needed.

Everything else looks good. Thanks for reviewing and updating the other requirements.
Thanks --Andy

ztucker4 commented:
@akohler, this is ready for a second look. I fixed the [QDB Reports](http://localhost:8000/qdb/report/) logout link, and I replaced the AUL Consales report with one for AUL Gomez. I think this is all of the code changes needed to support Alli's request, but let me know if I missed anything.

akohler reviewed (approved):
@ztucker4 All looks good now, thanks.  Merging. -Andy
